src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -346,37 +346,17 @@ private List<Type> CreateInheritanceHierarchyList(Type type)
 
         private Name GetName(string p)
         {
-            if (p == null)
-            {
-                p = string.Empty;
-            }
-            return GetName(p, _nameManager);
+            return _nameManager.Add(p ?? "");
         }
 
         /////////////////////////////////////////////////////////////////////////////////
 
         private Name GetName(Type type)
         {
             string name = type.Name;
-            if (type.IsGenericType)
-            {
-                // Trim the name to remove the ` at the end.
-                name = name.Split('`')[0];
-            }
-            return GetName(name, _nameManager);
+            return type.IsGenericType ? _nameManager.Add(name, name.IndexOf('`')) : _nameManager.Add(name);
```

Comment:
I've a feeling it's something that earned its keep a lot more before being ported.